ft_substr.c

```diff
@@ -0,0 +1,24 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   ft_substr.c                                        :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: bchristo <bchristo@student.42wolfsburg.    +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2021/05/25 19:38:58 by bchristo          #+#    #+#             */
+/*   Updated: 2021/05/26 10:46:41 by bchristo         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "libft.h"
+
+char	*ft_substr(char const *s, unsigned int start, size_t len)
+{
+	void	*sub;
+
+	sub = malloc(++len * sizeof(char));
```

Comment:
Same here. Sometimes this willl allocate more memory than needed.

Reply:
same here, tried altering it and breaks everything but like split both use srtlcpy and I am almost sure that's how it is intended to be used? looking into it and coming back with a definitive answer!

Reply:
so I am using `strlen` to get the length and in the man says 
```
DESCRIPTION
       The strlen() function calculates the length of the string pointed to by s, excluding the terminating null byte ('\0').
```
so I assume that's why it needs to ++

Reply:
I am answering here but I think this is the reason u can not find the termination of arrays `dest[n] = 0;` 
`strlcpy` is doing it behind the scenes `split > substr > strlcpy

Reply:
https://stackoverflow.com/questions/9440834/why-is-the-argument-to-malloc-i1-in-this-c-code#:~:text=The%20%2B%201%20is%20to%20make,to%20change%20in%20the%20future.